backend/prompt_studio/prompt_studio_output_manager/output_manager_helper.py

```diff
@@ -144,3 +145,37 @@ def fetch_default_llm_profile(tool: CustomTool) -> ProfileManager:
             return ProfileManager.get_default_llm_profile(tool=tool)
         except DefaultProfileError:
             raise DefaultProfileError("Default ProfileManager does not exist.")
+
+    @staticmethod
+    def fetch_default_response(
+        tool_studio_prompts: list[ToolStudioPrompt], document_manager_id: str
+    ) -> dict[str, Any]:
+        # Initialize the result dictionary
+        result: dict[str, Any] = {}
+        # Iterate over ToolStudioPrompt records
+        for tool_prompt in tool_studio_prompts:
+            prompt_id = str(tool_prompt.prompt_id)
+            profile_manager_id = str(tool_prompt.profile_manager.profile_id)
+
+            # If profile_manager is not set, skip this record
+            if not profile_manager_id:
```

Comment:
@chandrasekharan-zipstack when we delete a default profile of prompt then it will be null. Also when user try to use combined output only creating prompt and without profile, in that case this check needed.

Reply:
@jagadeeswaran-zipstack 
1. When we enforce a constraint that the user can upload documents / add prompts only after adding a profile which is marked as default - we should not let them delete the default profile at all. Please ensure we note / handle this case elegantly
1. I don't clearly understand this combined output case you're mentioning - is it single pass? Ideally when a prompt is being added without the user explicitly mentioning any profile - we should take the default profile and use that.

This will ensure that we always have a profile associated with a prompt eliminating the need for such checks in all our logic